File: evals/refs/bs.zig
const std = @import("std");

fn erfAS(x: f64) f64 {
    const p = 0.3275911;
    const a1 = 0.254829592;
    const a2 = -0.284496736;
    const a3 = 1.421413741;
    const a4 = -1.453152027;
    const a5 = 1.061405429;
    const sign: f64 = if (x < 0.0) -1.0 else 1.0;
    const xa = @abs(x);
    const t = 1.0 / (1.0 + p * xa);
    const y = 1.0 - (((((a5*t + a4)*t) + a3)*t + a2)*t + a1) * t * @exp(-xa*xa);
    return sign * y;
}
fn ncdf(x: f64) f64 {
    return 0.5 * (1.0 + erfAS(x / @sqrt(@as(f64, 2.0))));
}

pub fn main(init: std.process.Init) !void {
    const arena = init.arena.allocator();
    var in_buf: [4096]u8 = undefined;
    var fr = std.Io.File.stdin().reader(init.io, &in_buf);
    const r = &fr.interface;
    var data: std.ArrayList(u8) = .empty;
    try r.appendRemainingUnlimited(arena, &data);
    var out_buf: [4096]u8 = undefined;
    var fw = std.Io.File.stdout().writer(init.io, &out_buf);
    const w = &fw.interface;
    var it = std.mem.tokenizeAny(u8, data.items, " \n\r\t");
    const t = try std.fmt.parseInt(usize, it.next().?, 10);
    var c: usize = 0;
    while (c < t) : (c += 1) {
        const S = try std.fmt.parseFloat(f64, it.next().?);
        const K = try std.fmt.parseFloat(f64, it.next().?);
        const Ty = try std.fmt.parseFloat(f64, it.next().?);
        const rate = try std.fmt.parseFloat(f64, it.next().?);
        const sigma = try std.fmt.parseFloat(f64, it.next().?);
        const d1 = (@log(S / K) + (rate + 0.5 * sigma * sigma) * Ty) / (sigma * @sqrt(Ty));
        const d2 = d1 - sigma * @sqrt(Ty);
        const price = S * ncdf(d1) - K * @exp(-rate * Ty) * ncdf(d2);
        try w.print("{d:.4}\n", .{price});
    }
    try w.flush();
}

File: evals/refs/bs.rs
use std::io::{self, Read, Write, BufWriter};

fn erf_as(x: f64) -> f64 {
    let p = 0.3275911;
    let a1 = 0.254829592;
    let a2 = -0.284496736;
    let a3 = 1.421413741;
    let a4 = -1.453152027;
    let a5 = 1.061405429;
    let sign = if x < 0.0 { -1.0 } else { 1.0 };
    let xa = x.abs();
    let t = 1.0 / (1.0 + p * xa);
    let y = 1.0 - (((((a5*t + a4)*t) + a3)*t + a2)*t + a1) * t * (-xa*xa).exp();
    sign * y
}

fn ncdf(x: f64) -> f64 {
    0.5 * (1.0 + erf_as(x / 2f64.sqrt()))
}

fn main() {
    let mut s = String::new();
    io::stdin().read_to_string(&mut s).unwrap();
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    let mut it = s.split_ascii_whitespace();
    let t: usize = it.next().unwrap().parse().unwrap();
    for _ in 0..t {
        let s_v: f64 = it.next().unwrap().parse().unwrap();
        let k: f64 = it.next().unwrap().parse().unwrap();
        let ty: f64 = it.next().unwrap().parse().unwrap();
        let r: f64 = it.next().unwrap().parse().unwrap();
        let sig: f64 = it.next().unwrap().parse().unwrap();
        let d1 = ((s_v/k).ln() + (r + 0.5*sig*sig)*ty) / (sig * ty.sqrt());
        let d2 = d1 - sig * ty.sqrt();
        let c = s_v * ncdf(d1) - k * (-r*ty).exp() * ncdf(d2);
        writeln!(out, "{:.4}", c).unwrap();
    }
}
